### DISCUSSION TITLE
i18n: How to use same defaultLocale for 2 different domains?

### DISCUSSION THREAD

**Discussion by kvndevils** [vercel/next.js#37991] [Help]
Hi all,

I have a domain name with a .com and .fr. 
I would like to define the defaultLocale to be the same for both. 
How would you achieve that please? I'm struggling with this one.

Here is the faulty config:

```
i18n: {
        locales: ['en', 'fr'],
        defaultLocale: 'en',
        localeDetection: false,
        domains: [
            {
                domain: 'domain1.com',
                defaultLocale: 'en'
            },
            {
                domain: 'domain2.fr',
                defaultLocale: 'fr'
            },
            {
                domain: 'domain2.com',
                defaultLocale: 'fr'
            }
        ]
    },
```

And here is the error I'm getting:
Both domain2.fr and domain2.com configured the defaultLocale fr but only one can. Change one item's default locale to continue

For now, I decided to remove domain2.com and set defaultLocale as 'fr' but it's not perfect as www.domain1.com uses 'fr' and domain1.com used 'en' as a result

Thank you in advance for the help!

**Comment by falsepopsky** [comment]
I think this should work?

[Domain Routing](https://nextjs.org/docs/advanced-features/i18n-routing#domain-routing)

```
i18n: {
        locales: ['en', 'fr'],
        defaultLocale: 'en',
        localeDetection: false,
        domains: [
            {
                domain: 'domain1.com',
                defaultLocale: 'en'
                locales: ['fr'],
            },
            {
                domain: 'domain2.fr',
                defaultLocale: 'fr'
            },
        ]
    },

```

**Comment by kvndevils** [comment]
ah thanks but no I do need the domain2.com to be set in 'fr'. It gets defaulted if I remove it.

**Comment by sandruky85** [comment]
Hi all, 
Do you have any solution for this? I'm having the same issue. I need to config multiple domains with the same defaultLocale but getting an error

**Comment by kvndevils** [comment]
So, here is what I ended up doing but it's definitely not a clean solution sadly ..
In my case, I have 3 domains: domain1.com (EN), domain2.fr (FR) and domain2.com (FR).

```
i18n: {
        locales: ['en', 'fr', 'en-GB'],
        defaultLocale: 'fr',
        localeDetection: false,
        domains: [
            {
                domain: 'www.domain1.com',
                defaultLocale: 'en-GB'
            },
            {
                domain: 'domain1.com',
                defaultLocale: 'en'
            },
            {
                domain: 'domain2.fr',
                defaultLocale: 'fr'
            }
        ]
    },
```

**Reply by sandruky85** [reply]
It doesn't help me. I have 3 domains for lang: 
i18n: {
        // These are all the locales you want to support in
        // your application
        locales: ['en', 'es', 'de'],
        // This is the default locale you want to be used when visiting
        // a non-locale prefixed path e.g. `/hello`
        defaultLocale: 'en',
        browserLanguageDetection: false,
        domains: [
            {
                // Note: subdomains must be included in the domain value to be matched
                // e.g. www.example.com should be used if that is the expected hostname
                domain: 'localhost',
                defaultLocale: 'en',
                locales: ['en', 'de', 'es''],
                http: true,
            },
            {
                domain: 'domain1.com',
                defaultLocale: 'es',
            },
            {
                domain: 'domain2.com',
                defaultLocale: 'es',
            },
            {
                domain: 'www.domain3.com',
                defaultLocale: 'es',
            },
            {
                domain: 'domain4.com',
                defaultLocale: 'de',
            },
            {
                domain: 'domain5.com',
                defaultLocale: 'de',
            },
            {
                domain: 'www.domain6.com',
                defaultLocale: 'de',
            },
        ],
    },

**Comment by kvndevils** [comment]
just create copies of your lng dictionaries and assign different names (e.g. de-xx)?

**Reply by rishadAlamy** [reply]
then how the i18n will find trans from de.json

**Comment by eric-burel** [comment]
Hi there is really no better solution here? I have the exact same use case of .fr and .com for a website in French + the default version in English on another domain

**Comment by sandruky85** [comment]
Hi there, I'm using the env variable as the default language for all the subdomains and using the `domains` key for the main ones. Ex:
`i18n: {
        locales: ['en', 'de', 'es'],
        defaultLocale: process.env.APP_LANG,
        localeDetection: false,
        domains: [
            {
                domain: 'domain.en',
                defaultLocale: 'en',
            },
            {
                domain: 'domain1.es',
                defaultLocale: 'es',
            },
            {
                domain: 'domain2.de',
                defaultLocale: 'de',
            },
        ],
    },`

Maybe is not the best solution, but it is working for me

**Comment by rishadAlamy** [comment]
sorry can you explain how you tackled this ?
